xarray/tests/test_units.py

```diff
@@ -1021,10 +1091,11 @@ def test_replication_dataarray(func, dtype):
     data_array = xr.DataArray(data=array, dims="x")
 
     numpy_func = getattr(np, func.__name__)
-    expected = xr.DataArray(data=numpy_func(array), dims="x")
+    units = extract_units(numpy_func(data_array))
```

Comment:
would it help to abstract this away to a function that
1. extracts units
2. applies a function to unit-less objects
3. reattaches units

Reply:
in general, yes, but there are three variations of this pattern:
1. one where we have to precompute using pint quantities to extract units
2. two where we have to convert before stripping and
  a. then reattach the extracted units
  b. then convert back to the old units

I think I would probably create a huge mess with that function, so it might be better to leave that to a new PR?

Reply:
:+1: to new PR but only if you think it will help.